content/posts/a_confusing_build_failure.md

```diff
@@ -9,56 +9,50 @@ tags: ["testing", "continuous integration"]
+So why was the build failing? When the unit tests ran some of them would fail with an error stating that Mockito (mockitokotlin](https://github.com/nhaarman/mockito-kotlin)) - a library commonly used in Java and Android to provide mock test doubles and behavior verification - cannot mock final classes.
 
-In order to understand the error let's try to understand first how Mockito works. It creates at runtime objects that extend interfaces or a classes which can be used to verify that some methods were called with specific parameters. If a class that is attempted to be mocked is declared as `final` then the error above will be thrown, as a final class cannot be extended at compile or run time. In a unit test a Kotlin class was mocked and as Kotlin classes are by default
-`final` unless explicitly declared `open`, the error above was thrown.
+In order to understand the error, let's try to understand first how Mockito works. It creates objects at runtime, that implement/extend interfaces or classes which can be used to verify that some methods were called with specific parameters. If a class that is attempted to be mocked is declared as `final` then the error above will be thrown, as a final class cannot be extended at compile time and thus at run time. In our case, there was a unit test in which a Kotlin class was mocked and as Kotlin classes are by default `final`, unless explicitly declared `open`, the error above was thrown.
 
 The culprit was found but we still have not figured out why the tests pass locally but fail on the CI. 
 
 ## Maybe Mockito can mock final classes after all
 
-If Mockito cannot mock final classes then why do local tests pass? Well sure there must be some way to mock a final class, and that's exactly what happened by adding in the gradle build file the `org.mockito:mockito-inline` dependency. 
+If Mockito cannot mock final classes, then why do local tests pass? Well sure there must be some way to mock a final class, and that's exactly what happened. By adding in the gradle build file the `org.mockito:mockito-inline` dependency. 
 
-The problem is that the failing module did not compile with the dependency above. Another submodule did compile it, but it was declared on its `build.gradle` as `testImplementation` not `testApi`, meaning that final classes could be mocked on that submodule only, not on the failing module.
+The problem is that the failing module did not compile with the dependency above, another submodule did. And since the dependency was declared on the latter�s `build.gradle` as `testImplementation`, not `testApi`, final classes could be mocked on that submodule exclusively, and not on the failing one.
```

Comment:
I'm thinking to rewrite this paragraph as well as the next one. It confused both reviewers.